mirrord/layer/src/debugger_ports.rs

```diff
@@ -194,23 +262,52 @@ impl DebuggerPorts {
     ///
     /// Log errors (like malformed env variables) but do not panic.
     pub fn from_env() -> Self {
-        let detected = env::var(MIRRORD_DETECT_DEBUGGER_PORT_ENV)
+        let detected: Vec<u16> =
+            match env::var(MIRRORD_DETECT_DEBUGGER_PORT_ENV)
+                .ok()
+                .and_then(|s| {
+                    DebuggerType::from_str(&s)
+                        .inspect_err(|e| {
+                            error!(
+                                "Failed to decode debugger type from {} env variable: {}",
+                                MIRRORD_DETECT_DEBUGGER_PORT_ENV, e
+                            )
+                        })
+                        .ok()
+                }) {
+                Some(debugger) => debugger
+                    .get_port(&std::env::args().collect::<Vec<_>>(), |name| {
+                        std::env::var(name).ok()
+                    }),
+                None => vec![],
+            };
+        if !detected.is_empty() {
+            let mut value = String::new();
+            detected
+                .iter()
+                .for_each(|port| value.push_str(format!("{port},").as_str()));
+            // remove trailing comma
+            value.pop();
```

Comment:
```suggestion
            let value = detected.iter().map(ToString::to_string).collect::<Vec<_>>().join(",");
```
Only a suggestion tho